bundles/CoreBundle/Command/ClassesDefinitionsBuildCommand.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Command;
+
+use Pimcore\Console\AbstractCommand;
+use Pimcore\DataObject\ClassBuilder\PHPClassWriterInterface;
+use Pimcore\DataObject\ClassBuilder\PHPFieldCollectionClassWriterInterface;
+use Pimcore\DataObject\ClassBuilder\PHPObjectBrickClassWriterInterface;
+use Pimcore\DataObject\ClassBuilder\PHPObjectBrickContainerClassWriterInterface;
+use Pimcore\Model\DataObject;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+class ClassesDefinitionsBuildCommand extends AbstractCommand
+{
+    public function __construct(
+        protected PHPClassWriterInterface $classWriter,
+        protected PHPFieldCollectionClassWriterInterface $collectionClassWriter,
+        protected PHPObjectBrickClassWriterInterface $brickClassWriter,
+        protected PHPObjectBrickContainerClassWriterInterface $brickContainerClassWriter,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('pimcore:build:classes')
+            ->setDescription(
+                'rebuilds php files for classes, field collections and object bricks based on updated var/classes/definition_*.php files'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $objectClassesFolder = PIMCORE_CLASS_DEFINITION_DIRECTORY;
+        $files = glob($objectClassesFolder.'/*.php');
+
+        $changes = [];
+
+        foreach ($files as $file) {
+            $class = include $file;
```

Comment:
Why a different approach for classes? 
It's required that the class definition is also in the `classes` table, so why don't use the listing here as well, as `writePHPClasses()` doesn't create the database entry for the class definition.

Reply:
That is intentional. This command should be able to run WITHOUT Database.

Reply:
Idea is to create all PHP Classes during a Docker build for example where no db is available

Reply:
Would that be a BC break? That is the question...

Reply:
Hmm, yes, probably, because of possible conditions ... forget about it, topic for XI then 😉